### ISSUE TITLE
Dashio knows that my dash button was pressed but request url does not work

### ISSUE THREAD

**Issue by Erdccc** [danimtb/dasshio#63]
I see the following in my log:

Request: http://hassio/homeassistant/api/services/switch/switch.firplace_fan

but if I copy that into my web browser it says that site can not be read.

Running HA v 0.81.6 with the Dashio plugin

**Comment by lukastillmann** [comment]
I have the same or a similar issue. Here is my log:

Log
```
2018-11-25 08:32:14,871 | INFO | Reading config file: /data/options.json
2018-11-25 08:32:14,878 | INFO | Starting sniffing...
2018-11-25 08:32:27,573 | INFO | dash button pressed!
2018-11-25 08:32:27,574 | INFO | Request: http://hassio/homeassistant/api/services/notify/pushbullet
2018-11-25 08:32:27,576 | ERROR | Unable to perform  request: Check [url], [body], [headers] and API password or                     [domain], [service] and [service_data] format.
Traceback (most recent call last):
  File "/dasshio.py", line 56, in arp_display
    button["service_data"]), headers={'x-ha-access': os.environ.get('HASSIO_TOKEN')})
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 342, in decode
    raise JSONDecodeError("Extra data", s, end)
json.decoder.JSONDecodeError: Extra data: line 1 column 8 (char 7)
2018-11-25 08:32:27,669 | INFO | Packet captured, waiting 20s ...
```

I am also curious about the request url, but maybe the problem is the missing API password?

Running Hass.io 0.82.0.

**Comment by pmontgo33** [comment]
Also having the same issue. Any updates on a fix?

**Comment by danimtb** [comment]
Hi guys,

Sorry for the late reply.

I think this is actually not an error on Dasshio's side but a wrong configuration from yours. Let me answer each of you as problems seem to be different.

@Erdccc:

> Request: http://hassio/homeassistant/api/services/switch/switch.firplace_fan

This request seems to be malformed. I think maybeyou are trying to use  `http://hassio/homeassistant/api/services/switch/switch.turn_on`

Let me indicate an example that should work for you (replace name and mac with your own ones):
```
{
  "timeout": 20,
  "buttons": [
  {
    "name": "Gillette",
    "address": "AC:63:BE:77:C4:0D",
    "domain": "switch",
    "service": "turn_on",
    "service_data": "{\"entity_id\": \"switch.firplace_fan\"}"
  }
```

@lukastillmann:

Could you please include the config of the button that is causing your problem? Looks like any of the fields is missing (url, headers, body or domain, service, service_data).

I you are trying to send a notification message something like this could work:
```
{
  "timeout": 20,
  "buttons": [
  {
    "name": "Gillette",
    "address": "AC:63:BE:77:C4:0D",
    "domain": "notify",
    "service": "pushbullet",
    "service_data": "{\"message\": \"This is a test message!\"}"
  }
```

@pmontgo33:

Could you give more information about your setup? What error are you seeing? Which is your configuration? Thanks

Hope this helps! 😄

**Comment by pmontgo33** [comment]
@danimtb here is my configuration:

```
{

  "buttons": [
    {
      "name": "Bedroom Light Toggle",
      "address": "40:B4:CD:1D:8B:3F",
      "domain": "switch",
      "service": "toggle",
      "service_data": "{\"entity_id\": \"switch.bedroom_light_switch\"}"
    }
  ]
}
```

And I am getting the same error that lukastillmann is getting where it sees the button press, but says unable to perform request. Everything was working fine, and I didn't make any changes to the configuration and it just stopped working. Could it be related to home assistant's new auth system?

**Comment by danimtb** [comment]
Thanks for the information. Seems the config is ok. It could be related to the auth system but I think it was fixed in #47. However I haven't tried it with latest home assistant.

Please indicate the Dasshio version and HA version you are using and I will investigate it further. Thanks!

**Comment by pmontgo33** [comment]
Dasshio version 0.2.0
Hassio version 0.85.0

**Comment by danimtb** [comment]
You are running an outdated version of Dasshio. Authentication was solved in latest 0.3.1 release. Please update Dasshio from the Hass.io addon page in your instance

**Comment by pmontgo33** [comment]
I guess I didn't have this repo added in the add-on store, so it did not prompt me for an update. I will update it now, and test it when I get home. Thanks!
